### ISSUE TITLE
Unification of Japanses terminology

### ISSUE THREAD

**Issue by nzw0301** [fchollet/keras-docs-ja#40]
I want to unify terminology for more useful docs.

I found some different written forms

- [x] Numpy Array -> Numpy配列
- [x] . -> ．
- [x] , -> ，
- [x] : -> ：
- [x] ; -> ；
- [x] ( ) -> （）
- [x] str -> 文字列
- [x] boolean -> 真理値
- [x] int -> 整数
- [x] float -> 浮動小数点数
- [x] optimizer -> 最適化（アルゴリズム）
- [x] Arguments -> 引数
- [x] input shape -> 入力のshape
- [x] output shape -> 出力のshape
- [x] Return  -> 戻り値
- [x] Sequential Model -> Sequentialモデル
- [x] Functional API -> Functional API 
- [x] Recurrent -> Recurrent
- [x] metrics -> 評価値
- [x] EOS -> です/ます
- [x] See something -> ~~を参照
- [x] whether insert space or not before/after syntax highlight in sentence -> No
- [x] Data augmentation -> データ拡張
- [x] objective -> 目的関数
- [x] loss function -> 損失関数
- [x] training -> 学習
- [x] testing -> テスト
- [x] validation -> 検証
- [x] index -> インデックス
- [x] target -> ターゲット
- [x] layer -> レイヤー
- [x] shape -> shape
- [x] numbers (1 or 一) -> 1
- [x] regularizer -> 正則化
- [x] nD tensor -> n階テンソル
- [x] Fuzz factor -> 微小量

For example, `shape` is translated to "形状", "形" and "型" for now.

I think that translation to follow [Japanses Python docs](http://docs.python.jp/3/library/index.html) is easy for built-in.

thanks.

**Comment by henry0312** [comment]
- It's better to use '．' and '，' , instead of '。' and '、'.  
  (note that those are double-byte character)
- Return is "戻り値"
- Arguments is "引数"

I think.

**Comment by fchollet** [comment]
Indeed. This is the consequence of many people working on the translation at once. Unification of terminology would be great to have!

**Comment by nzw0301** [comment]
I think 
- "(", ")", ";" and ":" are ascii strings (not double-byte character)
  - insert one space after ")", ";" and ":".
  - insert one space before "(".
- insert one space before/after syntax highlight in a sentence

**Comment by nzw0301** [comment]
How do we decide unification of terminology?

**Comment by henry0312** [comment]
>  "(", ")", ";" and ":" are ascii strings (not double-byte character) 

since we use basically double-byte character, it's better to use "，", "．" as double-byte character.

> insert one space after ")", ";" and ":".
> insert one space before "(".

Addtionaly, if "(" and ")" are double-byte character, we can ignore these.

**Comment by nzw0301** [comment]
> Addtionaly, if "(" and ")" are double-byte character, we can ignore these.

I'm concerned about being mixed "(", ")" in Python syntax with "（", "）",
but I think that it is easy to use double-byte character for translation.

Sounds good!

**Comment by henry0312** [comment]
You have a point.
It's bother to confirm which to use double-byte characters or not.
Go ahead 👍

**Comment by Wakamatz** [comment]
I think that we should discuss in Japanese when discussing about Japanese translation.
Allow me following in Japanese.

現在nzw0301さんのところでKeras2についての翻訳がだいぶ進んでいますが，未だに翻訳のガイドラインが定まっていないのは問題だと考えています．
このようなものは統一されていることが重要だと思うのでまずはガイドラインを作りましょう．
下記の内容でいかがでしょうか？
よろしければ「Guidlines.md」と「TranslationTable.md」でPRします．

他に追加すべき項目や修正・削除すべき箇所があればご指摘下さい．

* Guidlines.md

-------------------------------

# ガイドライン
- 翻訳対象は本文とコード中のコメントとする．
- 本文は敬体（です・ます調）とする．
- 句読点は「，．」を用いる．
- ''，""は基本的にそのままとする．強調の意味で使用されている場合のみ「」に変換する．（bckend.md3行目）
- 記号「，．（）？！：；」は全角とする．
- 文中のシンタックスハイライト（syntax highlight）の前後に空白は入れない． 
- 用語の訳は対訳表に従う．
- Kerasの頭文字は大文字とする．

※ 翻訳は英語から日本語へのただの変換作業ではなく，英文の意味を読み取り，日本語として表現する創作作業です．
英語の言い回しに引きずられることなく自然な日本語を目指しましょう．

-------------------------------

* TranslationTable.md

-------------------------------

# 原則
- 構文キーワードなどはそのまま英語表記とする．

# 対訳表

| English | 日本語
|:---|:---
| Arguments | 引数
| Data augumentation | データ拡張
| layer | レイヤー
| loss function | 損失関数
| Return | 戻り値
| Recurrent  | 再帰
| shape | 形状
| target | ターゲット
| testing | 検証
| training | 学習

※ 見つけやすいようにアルファベット順で列挙しています．
必要に応じて追記してください．

-------------------------------

**Comment by nzw0301** [comment]
Thank you for your comment.
I don't have enough time now, sorry for not discussion.

I think that `README.md` contains this guildline for easier access to it.

I guess that it is unnecessary to translate some technical terms and python/numpy/keras unique terms into ja for searchability.
e.g. `Recurrent` and `shape`.

And 検証 means validation, not testing.

---

コメントありがとうございます．
ここしばらく時間がないので，すぐに議論ができないことお許し下さい．

これらのガイドラインを含めるとしたらアクセス性のために`README.md`に追記すべきだと思っています．

検索性のためにpython/numpy/keras特有の単語は翻訳する必要は無いと考えています．
例えば，`Recurrent`や`shape`です．

検証はvalidationの訳であり，testingとは別だと思いました．

**Comment by Wakamatz** [comment]
> アクセス性のためにREADME.mdに追記すべき

それでは`README.md`に追記する形にしましょう．


> 検索性のためにpython/numpy/keras特有の単語は翻訳する必要は無いと考えています．

これは良い考えだと思うので対訳表の前に明記します．
また，線引を明確にするため原語表記のものについても対訳表に追記します．

対訳表の変更分は下記のとおり．

| English | 日本語 |
| :--- | :--- |
| recurrent | recurrent |
| shape | shape |
| testing | テスト |

問題があれば後で改めれば良いと思うので，ここまでの内容でkeras-2の方にPR上げておきます．
